src/main/java/org/assertj/core/internal/Objects.java

```diff
@@ -731,7 +732,23 @@ public boolean areEqualToComparingOnlyGivenFields(Object actual, Object other, S
 	return isEqualToComparingOnlyGivenFields(actual, other, fields).isFieldsNamesEmpty();
   }
 
-  public static class ByFieldsComparison {
+  /**
+   * Verifies that actual {@code Object}s has the same toString result as expected has.
+   *
+   * @param info     contains information about the assertion.
+   * @param actual   the actual {@code Object}.
+   * @param expected the given {@code Object}.
+   * @throws AssertionError if the actual {@code Object} has not an equal toString result as expected has {@code Object}.
+   * @throws java.lang.NullPointerException if the actual {@code Object} has no scheme.
+   * @throws java.lang.NullPointerException if the actual {@code Object} has no scheme.
+   */
+  public boolean assertHasToString(AssertionInfo info, Object actual, Object expected) {
+  assertNotNull(info, actual);
+  assertNotNull(info, expected);
+  return areEqual(actual.toString(), expected.toString());
```

Comment:
create an error specific to this assertion with a message like:
Expecting toString() of Actual to be 
  < foo> 
but was 
  <bar>